src/components/Messages/Chat.js

```diff
@@ -1,93 +1,113 @@
+    const dispatch = useDispatch();
+    const userId = useSelector((state) => state.user.data.id);
+    const participants = useSelector(
+        (state) => state.messages.activeChat.participants
+    );
+    let messages = useSelector((state) => state.messages.activeChat.messages);
+    const [newMessageBody, setNewMessageBody] = useState({
+        sender: userId,
+        conversationId,
+        message: '',
+    });
 
-    const createMessage = async (message) => {
-        if (!message) return;
-        const res = await service.createMessage(authUser.id, cid, message);
-        if (res.error) {
-            return setError(
-                'We ran into an issue sending your message. Please try again later.'
-            );
-        }
-        setSentMessage(true);
-        setMessage('');
+    const listenForNewMessagesOnSocket = () => {
+        socket.emit('JOIN_ROOM');
+        socket.on('NEW_MESSAGE', () => {
+            // Make a REST post request to update inbox and active chat.
+            dispatch(findMessagesByConversationThunk(conversationId));
+        });
     };
 
     useEffect(() => {
-        const findMyMessages = async () => {
-            const res = await service.findAllMessagesByConversation(uid, cid);
-            if (res.error) {
-                return setError(
-                    'We ran into an issue showing your messages. Please try again later.'
-                );
-            }
-            // Mark sender and receiver bubbles
-            res.forEach((message) => {
-                message._id = message.id;
-                message.id = authUser.id === message.sender ? 0 : 1;
-            });
-            setMessages(res);
-        };
-        findMyMessages();
+        listenForNewMessagesOnSocket();
+        dispatch(findMessagesByConversationThunk(conversationId));
         // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [sentMessage]);
+    }, []);
+
+    const sendMessage = (e) => {
+        e.preventDefault();
+        dispatch(sendMessageThunk(newMessageBody));
+    };
 
     return (
         <div>
-            <h1>Messages</h1>
-            {error && <AlertBox message={error}/>}
+            <Link
+                to={`/messages`}
+                className='mt-2 me-2 btn btn-large btn-primary border border-secondary fw-bolder rounded-pill'
+            >
+                <i className='fa-solid fa-circle-arrow-left text-white fs-6'></i>
+                <span>Back to Inbox</span>
+            </Link>
+            <p className={"mt-4 mb-4"}>
+                Participants:
+                {participants.map((participant) => (
+                    <span className='badge rounded-pill bg-light' key={participant.id}>
+            {participant.name || participant.firstName}
+          </span>
+                ))}
+            </p>
+            {messages.length < 1 && <AlertBox message={'No messages.'}/>}
             {messages &&
                 <ChatFeed
```

Comment:
Seems like you're only passing messages to the external library component. If we create a separate component that takes messages as props as abstracts away all this customization, this would make this external dependency more loosely coupled and this component a bit shorter in line length.